feat/18 Add source for single file

deepu9 opened the pull request:
Closes #18 

1. Added ability to download a file from an url.
2. Also, run the checksum validation on the file and
3. Finally move the downloaded file to a source folder.

mirkobrombin reviewed (approved):
LGTM

kbdharun reviewed (approved):
LGTM, thanks for your contribution.

Tested the changes locally and made a few improvements:

1. Improved Error handling for checksum in the source types.
2. Updated docs to reflect the changes.
3. Updated Gitignore file to not commit Vib generated/config files to the repository.
4. Update website's index example (unrelated to PR).

_Recipe used for testing:_

```
name: Testing recipe
id: test
stages:
  - id: build
    base: ghcr.io/vanilla-os/pico:main
    singlelayer: false
    labels:
      maintainer: Vanilla OS Contributors
    modules:
      - name: build-app
        type: shell
        source:
          type: file
          url: https://github.com/Vanilla-OS/Vib/releases/download/v0.7.3/vib
          checksum: 5c4dbe2bef7c23175ad1c8230840a6b3c91c3ab3ed1eb026d9bc4b6eb382333d
        commands:
          - ./vib -v 
```

deepu9 commented:
> LGTM, thanks for your contribution.
> 
> Tested the changes locally and made a few improvements:
> 
> 1. Improved Error handling for checksum in the source types.
> 2. Updated docs to reflect the changes.
> 3. Updated Gitignore file to not commit Vib generated/config files to the repository.
> 4. Update website's index example (unrelated to PR).
> 
> _Recipe used for testing:_
> 
> ```
> name: Testing recipe
> id: test
> stages:
>   - id: build
>     base: ghcr.io/vanilla-os/pico:main
>     singlelayer: false
>     labels:
>       maintainer: Vanilla OS Contributors
>     modules:
>       - name: build-app
>         type: shell
>         source:
>           type: file
>           url: https://github.com/Vanilla-OS/Vib/releases/download/v0.7.3/vib
>           checksum: 5c4dbe2bef7c23175ad1c8230840a6b3c91c3ab3ed1eb026d9bc4b6eb382333d
>         commands:
>           - ./vib -v 
> ```

Thanks @kbdharun. My apologies for not updating the docs.
